src/dls_bluesky_core/core/scanspecs.py

```diff
@@ -0,0 +1,32 @@
+from typing import List, Optional
+
+import numpy as np
+from scanspec.core import Frames
+from scanspec.specs import DURATION
+
+
+def get_duration(frames: List[Frames]) -> Optional[float]:
+    """
+    Returns the duration of a number of ScanSpec frames, if known and consistent.
+
+    Args:
+        frames (List[Frames]): A number of Frame objects
+
+    Raises:
+        ValueError: If any frames do not have a Duration defined.
+
+    Returns:
+        duration (float): if all frames have a consistent duration
+        None: otherwise
+
+    """
+    for fs in frames:
+        if DURATION in fs.axes():
```

Comment:
Should also catch the empty spec edge case

```suggestion
        if DURATION in fs.axes() and len(fs.midpoints[DURATION]) > 0:
```

Reply:
Can an empty Spec have DURATION In axes?

Reply:
...looks like it can't, but that looks like a nasty error message since it lets you create a spec but crashes on `frames()`.
@coretl do we need a fix upstream? If so what should it do?

```python
>>> Static.duration(0, 0).frames()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/vid18871/projects/blueapi/venv/lib64/python3.9/site-packages/scanspec/specs.py", line 74, in frames
    return Path(self.calculate()).consume()
  File "/home/vid18871/projects/blueapi/venv/lib64/python3.9/site-packages/scanspec/specs.py", line 548, in calculate
    return _dimensions_from_indexes(
  File "/home/vid18871/projects/blueapi/venv/lib64/python3.9/site-packages/scanspec/specs.py", line 438, in _dimensions_from_indexes
    gap[0] = gap_between_frames(dimension, dimension)
  File "/home/vid18871/projects/blueapi/venv/lib64/python3.9/site-packages/scanspec/core.py", line 465, in gap_between_frames
    return any(frames1.upper[a][-1] != frames2.lower[a][0] for a in frames1.axes())
  File "/home/vid18871/projects/blueapi/venv/lib64/python3.9/site-packages/scanspec/core.py", line 465, in <genexpr>
    return any(frames1.upper[a][-1] != frames2.lower[a][0] for a in frames1.axes())
IndexError: index -1 is out of bounds for axis 0 with size 0

```

Reply:
Not sure, is an empty Spec a valid thing?

Reply:
I think an empty `Spec` should be valid, but `Frames` should always have length >0, raising a suitable error in the constructor to say this would create an empty path